PYIC-8776: add new page for cross-browser problem

thebauSoftwire opened the pull request:
## Proposed changes
### What changed

- added new problem-different-browser page
- add support for routing from problem-different-browser page to returning to the RP

### Why did it change

- we currently silently return the user back to the RP in a cross-browser scenario and the user is given little or misleading information about why this happened. To help ease frustration, we show them this new error page just before they go back to the RP

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8776](https://govukverify.atlassian.net/browse/PYIC-8776)

## Checklists

- [ ] READMEs and documentation up-to-date
- [x] Browser/ unit/ Selenium tests have been written/ updated
- [x] No risk of exposure: PII, credentials, etc through logs/ code
- [x] Ensure added/updated routes have CSRF protection if required


[PYIC-8776]: https://govukverify.atlassian.net/browse/PYIC-8776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

thebauSoftwire committed:
PYIC-8776: add new cross-browser-problem.njk page

thebauSoftwire committed:
PYIC-8776: add screenshots for new cross-browser-problem.njk page

thebauSoftwire committed:
PYIC-8776: wrap continue button in form to trigger posting journey to back-end

thebauSoftwire committed:
PYIC-8776: add routing to handle routing to and from the cross-browser-problem page

thebauSoftwire committed:
PYIC-8776: make middleware for handling return to rp from cross-browser-problem page a bit stricter

thebauSoftwire committed:
PYIC-8776: add browser test to check that core-front calls core-back with the correct action

thebauSoftwire committed:
PYIC-8776: add more unit tests for new middleware

hugh-emerson commented on browser-tests/imposter/config/api-config.yaml:
```diff
@@ -688,3 +688,40 @@ resources:
+        operator: Contains
+    queryParams:
+      currentPage: "cross-browser-problem"
+    response:
+      statusCode: 200
+      # In reality, core-back will route to the RP, but
+      # for this test, we just want to check that
+      # core-front makes the appropriate request to core-back
+      content: |
+        {
+          "page": "page-multiple-doc-check"
+        }
```
Would it be clearer/more intuitive to do something like this? Maybe this is a misuse of "client" and there's something better though

```
          "client": {
            "redirectUrl": "https://example.com/return-to-rp"
          }
```

We seem to use example.com elsewhere in these tests so I've suggested that for consistency. However, I don't love the idea of the tests loading up a 3rd party site (example.com). Alternatively, can you do something like this?

```
"redirectUrl": "/page-ipv-success?test=crossBrowserProblem"
```

or even redirecting to something more clearly fake like /dev/template?

thebauSoftwire replied:
good point, thanks! I've also updated the existing client redirect test to not use example.com

thebauSoftwire committed:
PYIC-8776: update core-back impostor to return redirect rather than page for cross-browser-problem browser test

thebauSoftwire committed:
PYIC-8776: update existing client redirect browser test to not redirect to 3rd party site (example.com) and instead redirect to a dev template page

blakeyp reviewed (dismissed):
QA in dev before merging

hugh-emerson reviewed (dismissed):
👍

thebauSoftwire committed:
PYIC-8776: re-name cross-browser-problem to problem-different-browser

I had missed in the ACs that the page should have problem-different-browser as the ID. So I update the pageId in core-back as well as the associated events/journey URIs to reflect this
